stackslib/src/chainstate/coordinator/mod.rs

```diff
@@ -2212,9 +2251,40 @@ impl<
     }
 
     /// Outermost call to process a burnchain block.
+    /// Will call the Stacks 2.x or Nakamoto handler, depending on whether or not 
     /// Not called internally.
+    /// NOTE: the 2.x and Nakamoto handlers return `Some(..)` in _different_ circumstances.  If
+    /// that matters to you, then you should call them directly.
     pub fn handle_new_burnchain_block(&mut self) -> Result<Option<BlockHeaderHash>, Error> {
-        self.inner_handle_new_burnchain_block(&mut HashSet::new())
+        let canonical_burnchain_tip = self.burnchain_blocks_db.get_canonical_chain_tip()?;
+        let epochs = SortitionDB::get_stacks_epochs(self.sortition_db.conn())?;
+        let target_epoch_index = StacksEpoch::find_epoch(&epochs, canonical_burnchain_tip.block_height).expect("FATAL: epoch not defined for burnchain height");
+        let target_epoch = epochs.get(target_epoch_index).expect("FATAL: StacksEpoch::find_epoch() returned an invalid index");
+        if target_epoch.epoch_id < StacksEpochId::Epoch30 {
+            // burnchain has not yet advanced to epoch 3.0
+            self.handle_new_epoch2_burnchain_block(&mut HashSet::new())
```

Comment:
Rather than a nesting `if-else` here, can you just `return self.handle_new_epoch2_burnchain_block()` at this branch? That way, the function can just continue without an else.

Reply:
sure